include/click/glue.hh

```diff
@@ -528,10 +528,10 @@ typedef click_jiffies_t (*click_jiffies_fct_t)(void*);
 extern click_jiffies_fct_t click_jiffies_fct;
 extern void* click_jiffies_fct_data;
 #  define click_jiffies()      (click_jiffies_fct(click_jiffies_fct_data))
-#  define CLICK_HZ           1000
+#  define CLICK_HZ           1000000
```

Comment:
is it really necessary to change CLICK_HZ ? It seems like a major change that could break things. There might be places and external elements which assumes (it's bad but it could happen) that CLICK_HZ is 1000. In the Kernel mode it's even just the kernel timer ticks.
Timer themselves do not use jiffies. It's probably only used for simplification somewhere.

Reply:
The reason I changed it is that BandwidthRatedUnqueue puts the task to sleep when its TokenBucket is empty, and schedules a timer to wake it up once enough tokens have accumulated. If the timer resolution isn't high enough, it will sleep for too long, and cause huge latency spikes.
See [elements/standard/bwratedunqueue.cc:88](https://github.com/tbarbette/fastclick/blob/main/elements/standard/bwratedunqueue.cc#L88).

Reply:
Alternatively, I could implement a different helper class for TokenBucket to use microseconds/nanoseconds instead of jiffies.

Reply:
Yes I think instead of make_jiffies you can simply use make_usec